backend/imagekit/client/media.go

```diff
@@ -0,0 +1,345 @@
+type DeleteFolderParam struct {
+	FolderPath string `validate:"nonzero" json:"folderPath"`
+}
+
+// MoveFolderParam represents parameter to move folder api
+type MoveFolderParam struct {
+	SourceFolderPath string `validate:"nonzero" json:"sourceFolderPath"`
+	DestinationPath  string `validate:"nonzero" json:"destinationPath"`
+}
+
+// JobIdResponse respresents response struct with JobId for folder operations
+type JobIdResponse struct {
+	JobId string `json:"jobId"`
+}
+
+// // FolderResponse respresents struct for response to move folder api.
+// type FolderResponse struct {
+// 	Data JobIdResponse
+// 	Response
+// }
+
+// JobStatus represents response Data to job status api
+type JobStatus struct {
+	JobId  string `json:"jobId"`
+	Type   string `json:"type"`
+	Status string `json:"status"`
+}
+
+// // JobStatusResponse represents response to job status api
+// type JobStatusResponse struct {
+// 	Data JobStatus
+// 	Response
+// }
+
+// File represents media library File details.
+func (ik *ImageKit) File(ctx context.Context, fileId string) (*http.Response, *File, error) {
+	data := &File{}
+	response, err := ik.HttpClient.CallJSON(ctx, &rest.Opts{
+		Method:       "GET",
+		Path:         fmt.Sprintf("/files/%s/details", fileId),
+		RootURL:      ik.Prefix,
+		IgnoreStatus: true,
+	}, nil, data)
+
+	if err != nil {
+		return response, data, err
+	}
+
+	if response.StatusCode != 200 {
+		err = fmt.Errorf("http error %d: %v", response.StatusCode, response.Status)
+	}
+
+	return response, data, err
+}
+
+// Files retrieves media library files. Filter options can be supplied as FilesOrFolderParam.
+func (ik *ImageKit) Files(ctx context.Context, params FilesOrFolderParam, includeVersion bool) (*http.Response, *[]File, error) {
+	var SearchQuery string = `type = "file"`
+
+	if includeVersion {
+		SearchQuery = `type IN ["file", "file-version"]`
+	}
+	if params.SearchQuery != "" {
+		SearchQuery = params.SearchQuery
+	}
+
+	parameters := url.Values{}
+
+	parameters.Set("skip", fmt.Sprintf("%d", params.Skip))
+	parameters.Set("limit", fmt.Sprintf("%d", params.Limit))
+	parameters.Set("path", params.Path)
+	parameters.Set("searchQuery", SearchQuery)
+
+	data := &[]File{}
+
+	response, err := ik.HttpClient.CallJSON(ctx, &rest.Opts{
+		Method:     "GET",
+		Path:       "/files",
+		RootURL:    ik.Prefix,
+		Parameters: parameters,
+	}, nil, data)
+
+	if err != nil {
+		return response, data, err
+	}
+
+	if response.StatusCode != 200 {
+		err = ParseError(response)
+	}
+
+	return response, data, err
+}
+
+// DeleteFile removes file by FileId from media library
+func (ik *ImageKit) DeleteFile(ctx context.Context, fileId string) (*http.Response, error) {
+	var err error
+
+	if fileId == "" {
+		return nil, errors.New("fileId can not be empty")
+	}
+
+	response, err := ik.HttpClient.CallJSON(ctx, &rest.Opts{
+		Method:  "DELETE",
+		Path:    fmt.Sprintf("/files/%s", fileId),
+		RootURL: ik.Prefix,
+	}, nil, nil)
```

Comment:
Note that if you supply `nil` as the `response` to `CallJSON` you need to close the body, from the docs

> If response is nil then the resp.Body will be closed only if opts.NoResponse is set.

Can you double check all the calls.

Reply:
Verified and updated